com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/DiffusionProfileShaderProperty.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using UnityEditor.Graphing;
+using UnityEngine;
+using UnityEditor.ShaderGraph.Drawing;
+using UnityEditor.ShaderGraph.Drawing.Inspector.PropertyDrawers;
+using UnityEditor.ShaderGraph.Internal;
+using UnityEngine.Rendering.HighDefinition;
+using UnityEditor.ShaderGraph;
+using System.Globalization;
+using static UnityEditor.ShaderGraph.Drawing.Inspector.PropertyDrawers.ShaderInputPropertyDrawer;
+
+namespace UnityEditor.Rendering.HighDefinition
+{
+    [Serializable]
+    [BlackboardInputInfo(55)]
+    class DiffusionProfileShaderProperty : AbstractShaderProperty<LazyLoadReference<DiffusionProfileSettings>>, IShaderPropertyDrawer
+    {
+        internal DiffusionProfileShaderProperty()
+        {
+            displayName = "Diffusion Profile";
+        }
+
+        internal override bool isBatchable => true;
+        internal override bool isExposable => true;
+        internal override bool isRenamable => true;
+        internal override bool isGpuInstanceable => true;
+
+        public override PropertyType propertyType => PropertyType.Vector1;
+
+        string assetReferenceName => $"{referenceName}_Asset";
+
+        internal override string GetPropertyBlockString()
+        {
+            uint hash = 0;
+            Vector4 asset = Vector4.zero;
+
+            if (value.isSet)
+            {
+                hash = value.asset.profile.hash;
+                asset = HDUtils.ConvertGUIDToVector4(AssetDatabase.AssetPathToGUID(AssetDatabase.GetAssetPath(value.asset)));
+            }
+
+            /// <summary>Float to string convertion function without any loss of precision</summary>
+            string f2s(float f) => System.Convert.ToDouble(f).ToString("0." + new string('#', 339));
+
+            return
+$@"[DiffusionProfile]{referenceName}(""{displayName}"", Float) = {f2s(HDShadowUtils.Asfloat(hash))}
```

Comment:
note: We should move the function asFloat from HDShadowUtils, can be done in another PR